### DISCUSSION TITLE
[Errno 113] No route to host hangs in Home Assistant

### DISCUSSION THREAD

**Discussion by OStrama** [pymodbus-dev/pymodbus#2393] [Help] [answered]
I use ModbusTcpClient in my Home Assistant Integration https://github.com/OStrama/weishaupt_modbus
When the client hardware gets disconnected (i.e., plug off Ethernet cable), the complete Home Assistant gets blocked. After connecting the client again, it works again. I see two error messages in the log:

Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: pymodbus.logging
Quelle: custom_components/weishaupt_wbb/modbusobject.py:45
Integration: Weishaupt WBB (Dokumentation, Probleme)
Erstmals aufgetreten: 17. Oktober 2024 um 21:16:41 (31 Vorkommnisse)
Zuletzt protokolliert: 17. Oktober 2024 um 21:17:45

Connection to (192.168.42.144, 502) failed: timed out
Connection to (192.168.42.144, 502) failed: [Errno 113] No route to host

and

Logger: homeassistant
Quelle: /srv/homeassistant/lib/python3.12/site-packages/homeassistant/runner.py:148
Erstmals aufgetreten: 17. Oktober 2024 um 21:17:48 (1 Vorkommnisse)
Zuletzt protokolliert: 17. Oktober 2024 um 21:17:48

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.12/site-packages/aioshelly/rpc_device/wsrpc.py", line 266, in _ping_if_not_closed
    await self._client.ping()
  File "/srv/homeassistant/lib/python3.12/site-packages/aiohttp/client_ws.py", line 166, in ping
    await self._writer.ping(message)
  File "/srv/homeassistant/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 715, in ping
    await self._send_frame(message, WSMsgType.PING)
  File "/srv/homeassistant/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 675, in _send_frame
    self._write(header + mask + message)
  File "/srv/homeassistant/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 702, in _write
    raise ConnectionResetError("Cannot write to closing transport")
ConnectionResetError: Cannot write to closing transport

What I do in my code, is simply reading and writing single values in this class: https://github.com/OStrama/weishaupt_modbus/blob/main/custom_components/weishaupt_modbus/modbusobject.py

What do I do wrong? How can I avoid blocking?

Thanks in advance for your help!

**Comment by janiversen** [comment]
As you can see the shelly integration also have a problem, indicating this is a hass problem

Anyhow why report it here, this is not the homeassistant project, nor do we have anything to do with the homeassistant modbus integration (which for some reason you do not use)

**Comment by janiversen** [comment, marked as answer]
you are using the sync. client, so of course everything blocks, when a call is active.

Homeassistant also uses an old version of pymodbus which we do not support.
